js-pkg/client/src/provider.ts

```diff
@@ -163,56 +208,56 @@ export class YSweetProvider {
     }
   }
 
-  async connect() {
+  /**
+   * Attempts to connect to the websocket.
+   * Returns a promise that resolves to true if the connection was successful, or false if the connection failed.
+   */
+  private attemptToConnect(clientToken: ClientToken): Promise<boolean> {
+    let promise = new Promise<boolean>((resolve) => {
+      let statusListener = (event: YSweetStatus) => {
+        if (event === STATUS_CONNECTED) {
+          this.off(EVENT_CONNECTION_STATUS, statusListener)
+          resolve(true)
+        } else if (event === STATUS_ERROR) {
+          this.off(EVENT_CONNECTION_STATUS, statusListener)
+          resolve(false)
+        }
+      }
+
+      this.on(EVENT_CONNECTION_STATUS, statusListener)
+    })
+
+    this.setupWs(clientToken)
```

Comment:
Would you be opposed to inlining this? All the small methods that are only used once make the connection flow a bit difficult for me to follow.